server/sonar-server/src/main/java/org/sonar/server/batch/ProjectDataLoader.java

```diff
@@ -56,15 +57,16 @@ public ProjectDataLoader(DbClient dbClient, UserSession userSession) {
   }
 
   public ProjectRepositories load(ProjectDataQuery query) {
-    boolean hasScanPerm = userSession.hasGlobalPermission(GlobalPermissions.SCAN_EXECUTION);
-    checkPermission(query.isIssuesMode());
-
     DbSession session = dbClient.openSession(false);
     try {
       ProjectRepositories data = new ProjectRepositories();
       ComponentDto module = checkFoundWithOptional(dbClient.componentDao().selectByKey(session, query.getModuleKey()),
         "Project or module with key '%s' is not found", query.getModuleKey());
 
+      boolean hasScanPerm = userSession.hasGlobalOrProjectPermissionByUuid(SCAN_EXECUTION, module.projectUuid());
+      boolean hasPreviewPerm = userSession.hasGlobalPermission(PREVIEW_EXECUTION);
```

Comment:
For the moment, it's still preview. I've never see "issues mode" in server source code. But JulienH told me that it should be renamed.